components/director/internal/repo/list.go

```diff
@@ -67,6 +68,10 @@ func (l *universalLister) List(ctx context.Context, tenant string, dest Collecti
 	return l.unsafeList(ctx, dest, additionalConditions...)
 }
 
+func (l *universalLister) SetSelectedColumns(selectedColumns []string) {
+	l.selectedColumns = strings.Join(selectedColumns, ", ")
```

Comment:
Perhaps in order to avoid concurrency problems, a copy of the lister can be created, modified and returned